packages/trader/src/Modules/Trading/Components/Form/TradeParams/barrier.jsx

```diff
@@ -13,14 +14,26 @@ const Barrier = ({
     barrier_count,
     barrier_pipsize,
     current_focus,
+    proposal_info,
     duration_unit,
     is_minimized,
     is_absolute_only,
     onChange,
     setCurrentFocus,
     validation_errors,
 }) => {
+    const contract_type_info =
+        (proposal_info?.CALL?.spot && proposal_info?.CALL) ||
+        proposal_info?.PUT ||
+        (proposal_info?.NOTOUCH?.spot && proposal_info?.NOTOUCH) ||
+        proposal_info?.ONETOUCH;
+    const current_price = contract_type_info?.spot || '';
+    const barrier_price = contract_type_info?.barrier || '';
     const barrier_title = barrier_count === 1 ? localize('Barrier') : localize('Barriers');
```

Comment:
should be `barrier_count > 1` in case barrier_count = 0.

Reply:
@henry-deriv I need to check which values can be in `barrier_count` (we take them from connection). The original code was written for `barrier_count === 1` and `barrier_count === 2`

Reply:
We use `<Barrier />` component only for those trade types, which have `barrier` in `components `array in file _packages/shared/src/utils/constants/contract.tsx_ (see the screenshot below). So, `<Barrier />`  will be rendered only for _Higher/Lower_, _Touch/No Touch_, _Ends In/Ends Out_, _Stays In/Goes Out_ and for them `barrier_count `is 1 or 2. For other trade types, which include `barrier_count` but don't have `components: ['barrier']` we won't render `<Barrier />`. 
In my conclusion, I would like to say, that if `<Barrier /> `is rendered, then `barrier_count` is 1 or 2. Please, if you know some nuances, write it here, because I might be wrong as have less experience. 
<img width="911" alt="Screenshot 2023-05-11 at 10 35 52 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/16488810-047c-4985-8944-aa60162d8414">

Reply:
I think you are right. Can ignore my comment 👍